### PR TITLE
[Filter:*] Migrate subplugins to ext @open sesame 1/31 20:24

### PR DESCRIPTION
**Pull request by myungjoo**
Filter subplugins that have external dependencies are
fully detached from Filter:main now.
    
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

File: ext/nnstreamer/tensor_filter/meson.build [hunk 1]
```diff
@@ -0,0 +1,68 @@
+
+if have_tensorflow
+  tensor_filter_tfcore_lib_shared = shared_library('nnstreamer_filter_tensorflow',
+      [
+        'tensor_filter_tensorflow_core.cc',
+        'tensor_filter_tensorflow.c'
+      ],
+    cpp_args: ['-Wno-sign-compare'],
+    dependencies: [tf_dep, protobuf_dep, glib_dep, gst_dep, orc_dep, nnstreamer_dep],
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/meson.build]: orc_dep, is it necessary?

Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/meson.build]: That's legacy of the past where this depended on tensor_common.h (memcpy-orc).

Let's experiment it without orc-dep

Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/meson.build]: Being tried with the next PR: https://github.com/nnsuite/nnstreamer/pull/1091

File: ext/nnstreamer/tensor_filter/meson.build [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    install_dir: filter_subplugin_install_dir
+  )
+
+  tensor_filter_tfcore_lib_static = static_library('nnstreamer_filter_tensorflow',
+      [
+        'tensor_filter_tensorflow_core.cc',
+        'tensor_filter_tensorflow.c'
+      ],
+    cpp_args: ['-Wno-sign-compare'],
+    dependencies: [tf_dep, protobuf_dep, glib_dep, gst_dep, orc_dep, nnstreamer_dep],
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/meson.build]: Later work, 
1. About the dependency nnstreamer_dep, now we declared nnstreamer_dep with nnstreamer base dependencies (glib, gst, gst-video/audio, orc, ..). Is it necessary to add this base dependencies?
2. Do we need to build static library for sub-plugins?

File: ext/nnstreamer/tensor_filter/meson.build [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    include_directories: nnstreamer_inc,
+    install: true,
+    install_dir: nnstreamer_libdir
+  )
+
+  tensor_filter_tfcore_lib = tensor_filter_tfcore_lib_shared
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_filter/meson.build]: line 27~33 and 60~67, unnecessary declaration

File: ext/nnstreamer/tensor_filter/meson.build (hunk 2 above)
Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/meson.build]: 1: Nope. I don't think so.

2: for Android project, we have the needs, releasing a whole big statically built binary containing the whole pipeline, gst, gst-plugins, nns, subplugins, nnfws, and custom filters.

File: ext/nnstreamer/tensor_filter/meson.build (hunk 3 above)
Comment (myungjoo) [reply ext/nnstreamer/tensor_filter/meson.build]: Commit updated

**Pull request merged**